### DISCUSSION TITLE
Multiple Key support for gamepad bindings

### DISCUSSION THREAD

**Discussion by UnixNight** [ppy/osu#26350] [Ideas]
Trying to set up my controller for Mania to mirror DJMAX Respect V but I can only set one key per action for 5 keys. They have it set up so right pad and X/Square are the middle key. Would it be possible to implement multiple keys for the same control for Mania mode? I know some beatmaps would be literally impossible on Controller but idk it's fun using a pad

**Comment by peppy** [comment]
We can probably allow two bindings per action in mania.

**Reply by Zyfarok** [reply]
You probably should enforce the two keys to conflict (like how mouse/keyboard keys conflict in osu!) so that it is not possible/easy to abuse this on faster maps. (like jacks ?)
Would need opinion of more experienced mania players on that.

**Reply by eclecticfluff** [reply]
It would be nice to allow scratch lanes to use two binds without restrictions though, as I do believe that’s typically the intent for how they are played.

**Reply by peppy** [reply]
> You probably should enforce the two keys to conflict (like how mouse/keyboard keys conflict in osu!) so that it is not possible/easy to abuse this on faster maps. (like jacks ?) Would need opinion of more experienced mania players on that.

all gameplay bindings work this same way.

**Reply by Paperpull** [reply]
this is required for at least middle and scratch lanes because there are ranked maps with alternation (using 2 fingers for one lane) in mind like in NekovhParavh (Trance Edit) top diff 01:39:124 or Solar Storm 5K top diff 01:07:785 and it would be consistent with stable (scratch lanes have the choice to be bindable with 2 keys) but adding middle column would bring people from other vsrg like djmax, ez2on and kalpa
